// @formatjs/ecma402-abstract/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.invariant = exports.isMissingLocaleDataError = exports.unpackData = exports.getLocaleHierarchy = exports.defineProperty = exports.getMagnitude = exports.setMultiInternalSlots = exports.setInternalSlot = exports.isLiteralPart = exports.getMultiInternalSlots = exports.getInternalSlot = exports.parseDateTimeSkeleton = exports.DATE_TIME_PROPS = exports._formatToParts = exports.BestFitFormatMatcher = void 0;
var BestFitFormatMatcher_1 = require("./src/DateTimeFormat/BestFitFormatMatcher");
Object.defineProperty(exports, "BestFitFormatMatcher", { enumerable: true, get: function () { return BestFitFormatMatcher_1.BestFitFormatMatcher; } });
__exportStar(require("./src/CanonicalizeLocaleList"), exports);
__exportStar(require("./src/CanonicalizeTimeZoneName"), exports);
__exportStar(require("./src/DateTimeFormat/BasicFormatMatcher"), exports);
__exportStar(require("./src/DateTimeFormat/DateTimeStyleFormat"), exports);
__exportStar(require("./src/DateTimeFormat/FormatDateTime"), exports);
__exportStar(require("./src/DateTimeFormat/FormatDateTimeToParts"), exports);
__exportStar(require("./src/DateTimeFormat/InitializeDateTimeFormat"), exports);
__exportStar(require("./src/DateTimeFormat/PartitionDateTimePattern"), exports);
__exportStar(require("./src/DateTimeFormat/ToDateTimeOptions"), exports);
__exportStar(require("./src/DisplayNames/CanonicalCodeForDisplayNames"), exports);
__exportStar(require("./src/GetNumberOption"), exports);
__exportStar(require("./src/GetOption"), exports);
__exportStar(require("./src/IsSanctionedSimpleUnitIdentifier"), exports);
__exportStar(require("./src/IsValidTimeZoneName"), exports);
__exportStar(require("./src/IsWellFormedCurrencyCode"), exports);
__exportStar(require("./src/IsWellFormedUnitIdentifier"), exports);
__exportStar(require("./src/NumberFormat/ComputeExponent"), exports);
__exportStar(require("./src/NumberFormat/ComputeExponentForMagnitude"), exports);
__exportStar(require("./src/NumberFormat/CurrencyDigits"), exports);
__exportStar(require("./src/NumberFormat/FormatNumericToParts"), exports);
__exportStar(require("./src/NumberFormat/FormatNumericToString"), exports);
__exportStar(require("./src/NumberFormat/InitializeNumberFormat"), exports);
__exportStar(require("./src/NumberFormat/PartitionNumberPattern"), exports);
__exportStar(require("./src/NumberFormat/SetNumberFormatDigitOptions"), exports);
__exportStar(require("./src/NumberFormat/SetNumberFormatUnitOptions"), exports);
__exportStar(require("./src/NumberFormat/ToRawFixed"), exports);
__exportStar(require("./src/NumberFormat/ToRawPrecision"), exports);
__exportStar(require("./src/PartitionPattern"), exports);
__exportStar(require("./src/PluralRules/GetOperands"), exports);
__exportStar(require("./src/PluralRules/InitializePluralRules"), exports);
__exportStar(require("./src/PluralRules/ResolvePlural"), exports);
__exportStar(require("./src/RelativeTimeFormat/FormatRelativeTime"), exports);
__exportStar(require("./src/RelativeTimeFormat/FormatRelativeTimeToParts"), exports);
__exportStar(require("./src/RelativeTimeFormat/InitializeRelativeTimeFormat"), exports);
__exportStar(require("./src/RelativeTimeFormat/MakePartsList"), exports);
__exportStar(require("./src/RelativeTimeFormat/PartitionRelativeTimePattern"), exports);
__exportStar(require("./src/RelativeTimeFormat/SingularRelativeTimeUnit"), exports);
__exportStar(require("./src/ResolveLocale"), exports);
__exportStar(require("./src/SupportedLocales"), exports);
var format_to_parts_1 = require("./src/NumberFormat/format_to_parts");
Object.defineProperty(exports, "_formatToParts", { enumerable: true, get: function () { return __importDefault(format_to_parts_1).default; } });
var utils_1 = require("./src/DateTimeFormat/utils");
Object.defineProperty(exports, "DATE_TIME_PROPS", { enumerable: true, get: function () { return utils_1.DATE_TIME_PROPS; } });
var skeleton_1 = require("./src/DateTimeFormat/skeleton");
Object.defineProperty(exports, "parseDateTimeSkeleton", { enumerable: true, get: function () { return skeleton_1.parseDateTimeSkeleton; } });
var utils_2 = require("./src/utils");
Object.defineProperty(exports, "getInternalSlot", { enumerable: true, get: function () { return utils_2.getInternalSlot; } });
Object.defineProperty(exports, "getMultiInternalSlots", { enumerable: true, get: function () { return utils_2.getMultiInternalSlots; } });
Object.defineProperty(exports, "isLiteralPart", { enumerable: true, get: function () { return utils_2.isLiteralPart; } });
Object.defineProperty(exports, "setInternalSlot", { enumerable: true, get: function () { return utils_2.setInternalSlot; } });
Object.defineProperty(exports, "setMultiInternalSlots", { enumerable: true, get: function () { return utils_2.setMultiInternalSlots; } });
Object.defineProperty(exports, "getMagnitude", { enumerable: true, get: function () { return utils_2.getMagnitude; } });
Object.defineProperty(exports, "defineProperty", { enumerable: true, get: function () { return utils_2.defineProperty; } });
var data_1 = require("./src/data");
Object.defineProperty(exports, "getLocaleHierarchy", { enumerable: true, get: function () { return data_1.getLocaleHierarchy; } });
Object.defineProperty(exports, "unpackData", { enumerable: true, get: function () { return data_1.unpackData; } });
Object.defineProperty(exports, "isMissingLocaleDataError", { enumerable: true, get: function () { return data_1.isMissingLocaleDataError; } });
__exportStar(require("./types/relative-time"), exports);
__exportStar(require("./types/date-time"), exports);
__exportStar(require("./types/list"), exports);
__exportStar(require("./types/plural-rules"), exports);
__exportStar(require("./types/number"), exports);
__exportStar(require("./types/displaynames"), exports);
var utils_3 = require("./src/utils");
Object.defineProperty(exports, "invariant", { enumerable: true, get: function () { return utils_3.invariant; } });
__exportStar(require("./262"), exports);
